src/library/scala/concurrent/impl/ExecutionContextImpl.scala

```diff
@@ -131,21 +125,18 @@ private[concurrent] object ExecutionContextImpl {
     new ExecutionContextImpl(Option(es).getOrElse(createDefaultExecutorService(reporter)), reporter)
       with ExecutionContextExecutorService {
         final def asExecutorService: ExecutorService = executor.asInstanceOf[ExecutorService]
-        override def execute(command: Runnable) = executor.execute(command)
-        override def shutdown() { asExecutorService.shutdown() }
-        override def shutdownNow() = asExecutorService.shutdownNow()
-        override def isShutdown = asExecutorService.isShutdown
-        override def isTerminated = asExecutorService.isTerminated
-        override def awaitTermination(l: Long, timeUnit: TimeUnit) = asExecutorService.awaitTermination(l, timeUnit)
-        override def submit[T](callable: Callable[T]) = asExecutorService.submit(callable)
-        override def submit[T](runnable: Runnable, t: T) = asExecutorService.submit(runnable, t)
-        override def submit(runnable: Runnable) = asExecutorService.submit(runnable)
-        override def invokeAll[T](callables: Collection[_ <: Callable[T]]) = asExecutorService.invokeAll(callables)
-        override def invokeAll[T](callables: Collection[_ <: Callable[T]], l: Long, timeUnit: TimeUnit) = asExecutorService.invokeAll(callables, l, timeUnit)
-        override def invokeAny[T](callables: Collection[_ <: Callable[T]]) = asExecutorService.invokeAny(callables)
-        override def invokeAny[T](callables: Collection[_ <: Callable[T]], l: Long, timeUnit: TimeUnit) = asExecutorService.invokeAny(callables, l, timeUnit)
+        final override def shutdown() { asExecutorService.shutdown() }
+        final override def shutdownNow() = asExecutorService.shutdownNow()
+        final override def isShutdown = asExecutorService.isShutdown
+        final override def isTerminated = asExecutorService.isTerminated
+        final override def awaitTermination(l: Long, timeUnit: TimeUnit) = asExecutorService.awaitTermination(l, timeUnit)
+        final override def submit[T](callable: Callable[T]) = asExecutorService.submit(callable)
+        final override def submit[T](runnable: Runnable, t: T) = asExecutorService.submit(runnable, t)
+        final override def submit(runnable: Runnable) = asExecutorService.submit(runnable)
+        final override def invokeAll[T](callables: Collection[_ <: Callable[T]]) = asExecutorService.invokeAll(callables)
+        final override def invokeAll[T](callables: Collection[_ <: Callable[T]], l: Long, timeUnit: TimeUnit) = asExecutorService.invokeAll(callables, l, timeUnit)
+        final override def invokeAny[T](callables: Collection[_ <: Callable[T]]) = asExecutorService.invokeAny(callables)
+        final override def invokeAny[T](callables: Collection[_ <: Callable[T]], l: Long, timeUnit: TimeUnit) = asExecutorService.invokeAny(callables, l, timeUnit)
```

Comment:
What does `final` add if it's an anonymous class anyway?

Reply:
@NthPortal In case it is moved out, I want(ed) to make sure that it stays final.

Reply:
fair enough 👍